node/src/service.rs

```diff
@@ -71,12 +75,44 @@ pub fn new_partial(
         .transpose()?;
 
     let executor = sc_service::new_native_or_wasm_executor(config);
+    let mut inherent_data = InherentData::new();
+
+    let timestamp = std::time::SystemTime::now()
+        .duration_since(std::time::SystemTime::UNIX_EPOCH)
+        .expect("Time is always after UNIX_EPOCH; qed")
+        .as_millis() as u64;
+
+    inherent_data
+        .put_data(sp_timestamp::INHERENT_IDENTIFIER, &timestamp)
+        .expect("Put timestamp inherent data");
+
+    let extrinsics: Vec<<Block as BlockT>::Extrinsic> =
```

Comment:
This is not a clean way to "cross the wasm boudary". The code you are writing here is on the client side. And you are calling `create_inherents` which is in the runtime. These kinds of calls should ideally go through a runtime api.

Reply:
As we discussed in element today, this is harder than I initially thought. It seems like genesis building may still be tied to the native runtime. I've asked about this in https://substrate.stackexchange.com/a/10128/372

So I'm okay with you calling directly into the runtime. But still, creating the extrinsics should not happen in the service file. The should happen in the genesis block builder implementation.